dask/bytes/core.py

```diff
@@ -0,0 +1,164 @@
+
+    Parameters
+    ----------
+    path: string
+    delimiter: bytes
+        An optional delimiter, like ``b'\n'`` on which to split blocks of bytes
+    not_zero: force seek of start-of-file delimiter, discarding header
+    blocksize: int (=128MB)
+        Chunk size
+    compression: string or None
+        String like 'gzip' or 'xz'.  Must support efficient random access.
+    sample: bool, int
+        Whether or not to return a sample from the first 10k bytes
+    **kwargs: dict
+        Options to send down to backend.  Includes authentication information
+        for systems like S3 or HDFS
+
+    Examples
+    --------
+    >>> sample, blocks = read_bytes('2015-*-*.csv', delimiter=b'\\n')  # doctest: +SKIP
+    >>> sample, blocks = read_bytes('s3://2015-*-*.csv', delimiter=b'\\n')  # doctest: +SKIP
+
+    Returns
+    -------
+    10kB sample header and list of ``dask.Delayed`` objects or list of lists of
+    delayed objects if ``fn`` is a globstring.
+    """
+    if compression is not None and compression not in compress_files:
+        raise ValueError("Compression type %s not supported" % compression)
+
+    if '://' in path:
+        protocol, path = path.split('://', 1)
+        try:
+            read_bytes = _read_bytes[protocol]
+        except KeyError:
+            raise NotImplementedError("Unknown protocol %s://%s" %
+                                      (protocol, path))
+    else:
+        read_bytes = _read_bytes['file']
+
+    return read_bytes(path, delimiter=delimiter, not_zero=not_zero,
+            blocksize=blocksize, sample=sample, compression=compression,
+            **kwargs)
+
+
+def open_files(path, compression=None, **kwargs):
+    """ Given path return dask.delayed file-like objects
+
+    Parameters
+    ----------
+    path: string
+        Filename or globstring
+    compression: string
+        Compression to use.  See ``dask.bytes.compression.files`` for options.
+    **kwargs: dict
+        Options to pass to storage backend.  Often used to pass authentication
+        information to S3 or HDFS.
+
+    Examples
+    --------
+    >>> files = open_files('2015-*-*.csv')  # doctest: +SKIP
+    >>> files = open_files('s3://2015-*-*.csv.gz', compression='gzip')  # doctest: +SKIP
+
+    Returns
+    -------
+    List of ``dask.delayed`` objects that compute to file-like objects
+    """
+    if compression is not None and compression not in compress_files:
+        raise ValueError("Compression type %s not supported" % compression)
+
+    if '://' in path:
+        protocol, path = path.split('://', 1)
+    else:
+        protocol = 'file'
+
+    try:
+        files = _open_files[protocol](path, **kwargs)
+    except KeyError:
+        raise NotImplementedError("Unknown protocol %s://%s" %
+                                  (protocol, path))
+    if compression:
+        decompress = merge(seekable_files, compress_files)[compression]
+        if sys.version_info[0] < 3:
```

Comment:
There are a few places where you do this check. It might be nice to implement `PY3` from compatibility, if for nothing else a slightly simpler thing to grep for when looking for version specific things.